pkg/apis/core/validation/validation.go

```diff
@@ -3024,6 +3025,47 @@ func ValidatePod(pod *core.Pod) field.ErrorList {
 		allErrs = append(allErrs, field.Invalid(specPath, hugePageResources, "must use a single hugepage size in a pod spec"))
 	}
 
+	podIPsField := field.NewPath("status", "podIPs")
+
+	// all PodIPs must be valid IPs
+	for i, podIP := range pod.Status.PodIPs {
+		for _, msg := range validation.IsValidIP(podIP.IP) {
+			allErrs = append(allErrs, field.Invalid(podIPsField.Index(i), podIP.IP, msg))
+		}
+	}
+
+	// if we have more than one Pod.PodIP then
+	// - validate for dual stack
+	// - validate for duplication
+	if len(pod.Status.PodIPs) > 1 {
+		// dualstack feature must be enabled if we have more than one PodIP
+		if !utilfeature.DefaultFeatureGate.Enabled(features.IPv6DualStack) {
```

Comment:
As discussed on slack, no need to check gates here and in CIDRs

Reply:
So this change - `dropDisabledFields` has been implemented on its separate commit (without removing the gate from validation). The concern i have is we removed the gated validated we won't be able to tell the user if they used dual stack features (multiple pod ips via CNI and/or multiple PodCIDRs)  without turning the feature gate on. Without the gated validation will just silently trim the slice to single entry. 

Am i not getting this correctly?

Reply:
@liggitt can you help by commenting on the above? ^^^
We want to move to `dropDisabledFields` style, where we trim PodIPs to PodIPs[0] when the gate is not enabled. The concern i have is detailed above, please let me know if it is not clear.

Reply:
We discussed this and agreed to drop